### PR TITLE
Update api.md

### PR DESCRIPTION
**Pull request by dgsponer**
reorder api endpoints
add storage_paths
add permission options

## Proposed change

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other (please explain): add documentation

## Checklist:

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [x] If applicable, I have included testing coverage for new code in this PR, for [backend](https://docs.paperless-ngx.com/development/#testing) and / or [front-end](https://docs.paperless-ngx.com/development/#testing-and-code-style) changes.
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [x] If applicable, I have checked that all tests pass, see [documentation](https://docs.paperless-ngx.com/development/#back-end-development).
- [x] I have run all `pre-commit` hooks, see [documentation](https://docs.paperless-ngx.com/development/#code-formatting-with-pre-commit-hooks).
- [x] I have made corresponding changes to the documentation as needed.
- [x] I have checked my modifications for any breaking changes.

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -321,6 +322,133 @@ granted). You can pass the parameter `full_perms=true` to API calls to view the
 full permissions of objects in a format that mirrors the `set_permissions`
 parameter above.
 
+## Permission options
```
Comment (shamoon) [inline docs/api.md:325]: Thanks for the PR. My only concern is going to be a bit of a chore to maintain and Im not sure how helpful it is as just a straight list. Hmm...

Comment (shamoon) [reply docs/api.md:325]: We also dont use all of them in a meaningful way so again, Im not sure of the purpose of an exhaustive list.

Comment (dgsponer) [reply docs/api.md:325]: Hi shamoon,

i work on an ansible-collection for configure paperless-ngx with ansible. creation tags, corresponders and as well users and so far. The sense was to serve a list with all possible options.

Comment (dgsponer) [reply docs/api.md:325]: the values coming from the api from the paperless default user, if you think, it's useless, no worry

**Pull request closed without merging**